src/Infrastructure.VS/VsInfoService.cs

```diff
@@ -0,0 +1,71 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using SonarLint.VisualStudio.Integration;
+
+namespace SonarLint.VisualStudio.Infrastructure.VS
+{
+    /// <summary>
+    /// Returns information about the current VS installation
+    /// </summary>
+    public interface IVsInfoService
+    {
+        /// <summary>
+        /// Returns the root directory for executing VS installation
+        /// </summary>
+        string InstallRootDir { get; }
+    }
+
+    [Export(typeof(IVsInfoService))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class VsInfoService : IVsInfoService
+    {
+        private readonly ILogger logger;
+
+        [ImportingConstructor]
+        public VsInfoService([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider, ILogger logger)
+        {
+            this.logger = logger;
+            InstallRootDir = GetInstallRootDir(serviceProvider);
+        }
+
+        public string InstallRootDir { get; }
+
+        private string GetInstallRootDir(IServiceProvider serviceProvider)
+        {
+            IVsShell shell = serviceProvider.GetService(typeof(SVsShell)) as IVsShell;
```

Comment:
I didn't bother handling the case where we can't fetch `IVsShell` - there would have to be something seriously wrong with VS for that to happen in practice, and it isn't something we could really recover from.